python/cuml/dask/preprocessing/LabelEncoder.py

```diff
@@ -0,0 +1,190 @@
+
+    def fit(self, y):
+        """
+        Fit a LabelEncoder (nvcategory) instance to a set of categories
+
+        Parameters
+        ----------
+        y : dask_cudf.Series
+            Series containing the categories to be encoded. It's elements
+            may or may not be unique
+
+        Returns
+        -------
+        self : LabelEncoder
+            A fitted instance of itself to allow method chaining
+
+        """
+        _classes = y.unique().compute()
+        el = first(y) if isinstance(y, Sequence) else y
+        self.datatype = ('cudf' if isinstance(el, (dcDataFrame, daskSeries))
+                         else 'cupy')
+        from cuml.preprocessing import LabelEncoder as LE
+        self._set_internal_model(LE(**self.kwargs).fit(y, _classes=_classes))
+        return self
+
+    def fit_transform(self, y, delayed=True):
+        """
+        Simultaneously fit and transform an input
+
+        This is functionally equivalent to (but faster than)
+        LabelEncoder().fit(y).transform(y)
+        """
+        return self.fit(y).transform(y, delayed=delayed)
+
+    def transform(self, y, delayed=True):
+        """
+        Transform an input into its categorical keys.
+
+        This is intended for use with small inputs relative to the size of the
+        dataset. For fitting and transforming an entire dataset, prefer
+        `fit_transform`.
+
+        Parameters
+        ----------
+        y : dask_cudf.Series
+            Input keys to be transformed. Its values should match the
+            categories given to `fit`
+
+        Returns
+        -------
+        encoded : cuPy backed Dask Array
+            The ordinally encoded input series
+
+        Raises
+        ------
+        KeyError
+            if a category appears that was not seen in `fit`
+        """
+        if self._get_internal_model() is not None:
+            return self._transform(y,
+                                   delayed=delayed,
+                                   output_dtype='int64',
+                                   output_collection_type='series')
+        else:
+            msg = ("This LabelEncoder instance is not fitted yet. Call 'fit' "
+                   "with appropriate arguments before using this estimator.")
+            raise NotFittedError(msg)
+
+    def inverse_transform(self, y, delayed=True):
+        """
+        Convert the data back to the original representation.
+        In case unknown categories are encountered (all zeros in the
+        one-hot encoding), ``None`` is used to represent this category.
+
+        Parameters
+        ----------
+        X : dask_cudf Series
+            The string representation of the categories.
+        delayed : bool (default = True)
+            Whether to execute as a delayed task or eager.
+
+        Returns
+        -------
+        X_tr : CuPy backed Dask Array
+            Distributed object containing the inverse transformed array.
```

Comment:
This describes the output as a cupy array, but below it seems to ask for cudf output type? Should make sure they are consistent and consistent with the single gpu version - apologies if I'm misunderstanding those conversions

Reply:
My bad, I made a mistake here.

Reply:
My bad, I made a mistake here.